distribution/openrct2.d.ts

```diff
@@ -387,96 +585,211 @@ declare global {
+         * 
+         * @return An array of Peeps.
+         */
         getAllEntities(type: "peep"): Peep[];
     }
 
+    /** Type of tile elements (e.g. surface, footpath, track). */
     type TileElementType =
         "surface" | "footpath" | "track" | "small_scenery" | "wall" | "entrance" | "large_scenery" | "banner"
-        /** This only exist to retrieve the types for existing corrupt elements. For hiding elements, use the isHidden field instead. */
+        /** This only exists to retrieve the types for existing corrupt elements. For hiding elements, use the isHidden field instead. */
         | "openrct2_corrupt_deprecated";
 
+    /**
+     * A tile element.
+     */
     interface BaseTileElement {
+        /** The tile type. */
         type: TileElementType;
+        /** The base height of the tile for collision. */
         baseHeight: number;
+        /** The top height of the tile for collision. */
         clearanceHeight: number;
-        isHidden: boolean; /** Take caution when changing this field, it may invalidate TileElements you have stored in your script. */
+        /** Indicates if the tile is hidden. Take caution when changing this field, it may invalidate 
+         * TileElements you have stored in your plugin. 
+         */
+        isHidden: boolean;
     }
 
+    /**
+     * A surface element.
+     */
     interface SurfaceElement extends BaseTileElement {
+        /** TODO what does the number mean? */
         slope: number;
+        /** A number indicating the surface style. */
         surfaceStyle: number;
+        /** A number indicating the edge style (blended or not?) TODO: verify */
         edgeStyle: number;
+        /** The surface height of the water. */
         waterHeight: number;
+        /** The length of the grass from X to Y. TODO */
         grassLength: number;
+        /** A number from x to y indicating the ownership status. TODO */
         ownership: number;
+        /** A number indicating the location of park fences. */
         parkFences: number;
 
+        /** True if the surface is owned by the park, false otherwise. */
         readonly hasOwnership: boolean;
+        /** True if the surface has construction rights owned by the park, false otherwise. */
         readonly hasConstructionRights: boolean;
     }
 
+    /**
+     * A footpath element.
+     */
     interface FootpathElement extends BaseTileElement {
+        /** A number indicating the footpath type. TODO */
         footpathType: number;
+        /** A number indicating the connection points on the path. TODO */
         edgesAndCorners: number;
+        /** A number between x and y indicating the slope direction, or null for a flat path. TODO */
         slopeDirection: number | null;
+        /** True if the path has a crossing which is occupied (e.g. by a train), false otherwise. */
         isBlockedByVehicle: boolean;
+        /** TODO? Is this for better pathfinding? */
```

Comment:
No its for bad pathfinding :P. This should never be adjusted as it is recalculated every couple ticks. Its used in the pathfinding to simplify the search space but its very hacky and one day will hopefully be removed.

Reply:
This would be a breaking change but should it ideally be `readonly`?

Reply:
Hey if people want to change the variable they are welcome to just don't expect anything useful to come from it.

Reply:
Not sure if I agree with that - making it readonly at least avoids the user wasting their time.